File: Kritika Tripathi/Basics Of Programming/Patterns/Pattern15.cpp
#include <iostream>
using namespace std;
int main(){
    int n;
    cin >> n;
    int scale = n/2+1;
    int j;
    //write your code here
    for(int i=1; i<=scale; i++){
        for(int k=1; k<=scale-i; k++){
            cout<<"\t";
        }
        int l = i;
        for(j=1; j<=i; j++,l++){
            cout<<l<<"\t";
        }
        l=2*i-2;
        for(; j<=2*i-1; j++,l--){
            cout<<l<<"\t";
        }
        cout<<endl;
    }
    for(int i=n; i>scale; i--){
        for(int k=1; k<=2*scale-i; k++){
            cout<<"\t";
        }
        int l = i-scale;
        for(j=1; j<=i-scale; j++,l++){
            cout<<l<<"\t";
        }
        l=2*(i-scale)-2;
        for(; j<=2*(i-scale)-1; j++,l--){
            cout<<l<<"\t";
        }
        cout<<endl;
    }
    
}

File: V.Vennela/Basics of Programming/Patterns/Pattern15.java
import java.util.*;

public class Pattern15 {

    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);

        // write ur code here
        int n = scn.nextInt();
        int sp=n/2,num=1,x=1;
        for(int i=1;i<=n;i++)
        {
            for(int j=1;j<=sp;j++)
            {
                System.out.print("\t");
            }
            int x1=x;
            for(int j=1;j<=num;j++)
            {
                System.out.print(x1+"\t");
                if(j<=num/2)
                {
                    x1++;
                }
                else
                {
                    x1--;
                }
            }
            if(i<=n/2)
            {
                sp--;
                num+=2;
                x++;
            }
            else
            {
                sp++;
                num-=2;
                x--;
            }
            System.out.println();
        }

    }
}
